rust-src/wallet_library/src/credential.rs

```diff
@@ -111,6 +117,102 @@ fn build_policy(
     })
 }
 
+/// The details for a new credential deployment.
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CredentialDeploymentDetails {
+    expiry:       TransactionTime,
+    unsigned_cdi: UnsignedCredentialDeploymentInfo<IpPairing, ArCurve, AttributeKind>,
+}
+
+/// Defines the byte serialization of the credential deployment details. Note
+/// that this serialization is for a new credential being deployed, which the
+/// explicit 0 byte denotes.
+impl Serial for CredentialDeploymentDetails {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        self.unsigned_cdi.serial(out);
+        0u8.serial(out);
+        self.expiry.seconds.serial(out);
+    }
+}
+
+/// Serializes the credential deployment details. It is this serialization that
+/// should be signed when creating a new credential for deployment. The
+/// serialized bytes are returned hex encoded.
+pub fn serialize_credential_deployment_details(
+    credential_deployment_details: CredentialDeploymentDetails,
+) -> String {
+    let mut serialized_details = Vec::<u8>::new();
+    credential_deployment_details.serial(&mut serialized_details);
+    hex::encode(serialized_details)
+}
+
+/// The required credential deployment context required to correctly serialize
+/// a new credential deployment so that it can be sent to the node.
+#[derive(SerdeSerialize, SerdeDeserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CredentialDeploymentPayload {
+    unsigned_cdi: UnsignedCredentialDeploymentInfo<IpPairing, ArCurve, AttributeKind>,
+    signatures:   Vec<String>,
+}
+
+/// Serializes the credential deployment payload. The result of this
+/// serialization should be sent as a raw payload to the node. The serialized
+/// bytes are returned hex encoded.
+pub fn serialize_credential_deployment_payload(
+    payload: CredentialDeploymentPayload,
+) -> Result<String> {
+    let cdi = get_credential_deployment_info(payload.signatures, payload.unsigned_cdi)?;
+    let acc_cred = AccountCredential::Normal { cdi };
+
+    let mut acc_cred_ser = Vec::<u8>::new();
+    acc_cred.serial(&mut acc_cred_ser);
+
+    Ok(hex::encode(acc_cred_ser))
+}
+
+fn get_credential_deployment_info(
+    signatures: Vec<String>,
+    unsigned_cdi: UnsignedCredentialDeploymentInfo<IpPairing, ArCurve, AttributeKind>,
+) -> Result<CredentialDeploymentInfo<IpPairing, ArCurve, AttributeKind>, anyhow::Error> {
+    let signature_map = build_signature_map(&signatures)?;
+    let proof_acc_sk = AccountOwnershipProof {
+        sigs: signature_map,
+    };
+
+    let cdp = CredDeploymentProofs {
+        id_proofs: unsigned_cdi.proofs,
+        proof_acc_sk,
+    };
+
+    Ok(CredentialDeploymentInfo {
+        values: unsigned_cdi.values,
+        proofs: cdp,
+    })
+}
+
+fn build_signature_map(
+    signatures: &[String],
+) -> Result<BTreeMap<KeyIndex, AccountOwnershipSignature>, anyhow::Error> {
```

Comment:
Can you elaborate on this function. It seems there are some implicit assumptions about indices and order for the input which would be good to document.

Reply:
I am removing this function and we will provide the map with keys directly instead.